static/js/services/contact-change-filter.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Service to identify relevant changes from Changes feed for contacts
+ */
+var _ = require('underscore');
+
+angular.module('inboxServices').factory('ContactChangeFilter',
+  function(ContactSchema) {
+    'ngInject';
+    'use strict';
+
+    var isValidInput = function(change, contact) {
+      return !!change && !!change.doc && !!contact && !!contact.doc;
+    };
+
+    var isContact = function(change) {
+      return !!change.doc.type && ContactSchema.getTypes().indexOf(change.doc.type) !== -1;
+    };
+
+    var isReport = function(change) {
+      return !!change.doc.form && change.doc.type === 'data_record';
+    };
+
+    var matchReportSubject = function(report, contact) {
+      if (report.doc.fields && (
+          (report.doc.fields.patient_uuid && report.doc.fields.patient_uuid === contact.doc._id) ||
+          (report.doc.fields.patient_id && report.doc.fields.patient_id === contact.doc.patient_id) ||
+          (report.doc.fields.place_id && report.doc.fields.place_id === contact.doc._id) ||
+          (report.doc.fields.place_id && report.doc.fields.place_id === contact.doc.place_id)
+        )) {
+        return true;
+      }
+
+      if ((report.doc.patient_id && report.doc.patient_id === contact.doc.patient_id) ||
+        (report.doc.place_id && report.doc.place_id === contact.doc.place_id)) {
+        return true;
+      }
+
+      return false;
+    };
+
+    var isChild = function(change, contact) {
+      return !!change.doc.parent && change.doc.parent._id === contact.doc._id;
+    };
+
+    var wasChild = function(change, contact) {
+      if (!contact.children) {
+        return false;
+      }
+      return !!_.find(contact.children, function(children) {
+        return children instanceof Array && _.find(children, function(child) {
```

Comment:
When is `children` not an Array?

Reply:
When `children` contain a `places` property, `children.childPlacesLabel` and `children.childPlacesIcon` are added, which are not arrays (see ContactViewModelGenerator).